File: src/lib/ipc/events.ts
import type { IpcEventMap } from '@/lib/types/ipc'
import { subscribeToEvent } from './client'

export function onDirPatch(handler: (payload: IpcEventMap['dir://patch']) => void) {
  return subscribeToEvent('dir://patch', handler)
}

export function onListChunk(handler: (payload: IpcEventMap['dir://list-chunk']) => void) {
  return subscribeToEvent('dir://list-chunk', handler)
}

export function onSizeState(handler: (payload: IpcEventMap['size://state']) => void) {
  return subscribeToEvent('size://state', handler)
}

export function onIconState(handler: (payload: IpcEventMap['icon://state']) => void) {
  return subscribeToEvent('icon://state', handler)
}

export function onVolumesChanged(handler: (payload: IpcEventMap['volumes://changed']) => void) {
  return subscribeToEvent('volumes://changed', handler)
}

export function onQueueProgress(handler: (payload: IpcEventMap['queue://progress']) => void) {
  return subscribeToEvent('queue://progress', handler)
}

export function onQueueConflict(handler: (payload: IpcEventMap['queue://conflict']) => void) {
  return subscribeToEvent('queue://conflict', handler)
}

export function onQueueRemoved(handler: (payload: IpcEventMap['queue://removed']) => void) {
  return subscribeToEvent('queue://removed', handler)
}

export function onWatchError(handler: (payload: IpcEventMap['watch://error']) => void) {
  return subscribeToEvent('watch://error', handler)
}

File: src-tauri/src/ipc/events.rs
pub const DIR_PATCH: &str = "dir://patch";
pub const SIZE_STATE: &str = "size://state";
pub const ICON_STATE: &str = "icon://state";
pub const ITEM_COUNT: &str = "item-count://state";
pub const VOLUMES_CHANGED: &str = "volumes://changed";
pub const QUEUE_PROGRESS: &str = "queue://progress";
pub const QUEUE_CONFLICT: &str = "queue://conflict";
pub const QUEUE_REMOVED: &str = "queue://removed";
pub const WATCH_ERROR: &str = "watch://error";
/// v2 directory-session patch event (Phase 5): carries a `SessionPatch`
/// (`delta`/`replaceView`/`metadataOnly`) instead of the legacy tab-scoped
/// `dir://patch` shape. Distinct event name so the un-migrated fraction of
/// the app (still on `DIR_PATCH`) and the v2 `PaneEntryCollection` consumer
/// never need to disambiguate a shared channel.
pub const DIR_SESSION_PATCH: &str = "dir://session-patch";
